driver/optmaindriver.ml

```diff
@@ -64,72 +63,73 @@ let main argv ppf =
         exit 2
       end
     end;
-    readenv ppf Before_link;
+    Compenv.readenv ppf Before_link;
     if
       List.length (List.filter (fun x -> !x)
                      [make_package; make_archive; shared;
-                      stop_early; output_c_object]) > 1
+                      Compenv.stop_early; output_c_object]) > 1
     then
     begin
       let module P = Clflags.Compiler_pass in
       match !stop_after with
       | None ->
-        fatal "Please specify at most one of -pack, -a, -shared, -c, \
-             -output-obj";
+          Compenv.fatal "Please specify at most one of -pack, -a, -shared, -c, \
+                         -output-obj";
       | Some ((P.Parsing | P.Typing | P.Scheduling) as p) ->
         assert (P.is_compilation_pass p);
-        Printf.ksprintf fatal
+        Printf.ksprintf Compenv.fatal
           "Options -i and -stop-after (%s) \
            are  incompatible with -pack, -a, -shared, -output-obj"
           (String.concat "|"
              (Clflags.Compiler_pass.available_pass_names ~native:true))
     end;
     if !make_archive then begin
       Compmisc.init_path ();
-      let target = extract_output !output_name in
+      let target = Compenv.extract_output !output_name in
       Asmlibrarian.create_archive
-        (get_objfiles ~with_ocamlparam:false) target;
+        (Compenv.get_objfiles ~with_ocamlparam:false) target;
       Warnings.check_fatal ();
     end
     else if !make_package then begin
       Compmisc.init_path ();
-      let target = extract_output !output_name in
+      let target = Compenv.extract_output !output_name in
       Compmisc.with_ppf_dump ~file_prefix:target (fun ppf_dump ->
         Asmpackager.package_files ~ppf_dump (Compmisc.initial_env ())
-          (get_objfiles ~with_ocamlparam:false) target ~backend);
+          (Compenv.get_objfiles ~with_ocamlparam:false) target ~backend);
       Warnings.check_fatal ();
     end
     else if !shared then begin
       Compmisc.init_path ();
-      let target = extract_output !output_name in
+      let target = Compenv.extract_output !output_name in
       Compmisc.with_ppf_dump ~file_prefix:target (fun ppf_dump ->
         Asmlink.link_shared ~ppf_dump
-          (get_objfiles ~with_ocamlparam:false) target);
+          (Compenv.get_objfiles ~with_ocamlparam:false) target);
       Warnings.check_fatal ();
     end
-    else if not !stop_early && !objfiles <> [] then begin
+    else if not !Compenv.stop_early && !objfiles <> [] then begin
       let target =
         if !output_c_object then
-          let s = extract_output !output_name in
+          let s = Compenv.extract_output !output_name in
           if (Filename.check_suffix s Config.ext_obj
             || Filename.check_suffix s Config.ext_dll)
           then s
           else
-            fatal
+          Compenv.fatal
```

Comment:
Nit: you've changed the indentation. I don't really care either way, but I'm pointing it out in case it wasn't on purpose.

Reply:
As far as I can tell that's what the `.ocp-indent` of the repo gives me.

Reply:
But it seems I didn't apply it constantly so I restored it.